### ISSUE TITLE
Missing Material#display_name=()

### ISSUE THREAD

**Issue by Fredosixx** [SketchUp/api-issue-tracker#780]
When creating a material, we can assign it a `name`, but not a `display_name`, which is what is displayed in the material browser.

For built-in materials and materials coming from skm files, the two properties are present.

It may be a good idea to have two separate names, one for display, one for programming. The main uage is that you would encode the `name` (with a pattern that can be recognized by a script), while letting users see a nice `display_name`.

So, could we have a method `Material#display_name=()`.

Thanks

**Comment by DanRathbun** [comment]
> The main uage is that you would encode the `name` (with a pattern that can be recognized by a script), while letting users see a nice `display_name`.

At one time there was a bug that did not save attribute dictionary to a SKM (or perhaps a material,) but I think this was fixed for quite some time.

How would encoding the `name` property be better than using a extension specific dictionary with attributes ?

And (more importantly) would encoding the `name` property interfere with it's role as the **key** for the materials collection ?


**I'm not against a setter for `display_name`, but what challenges would it cause ?**

One I can think of is finding a material by display name rather than by key `name`:
```ruby
module FindMaterialByDisplayName

  refine ::Sketchup::Materials do

    def get_by_display_name(arg)
      self.find {|matl| matl.display_name == arg }
    end

  end
end
```
The above would find the **first** material whose `display_name` matched, but there could be other matches as this property need not be unique across the collection.

----

On a sidenote, the doc snippet implies that internal material key names are delimited by `[ ]` brackets ?

But in reality there is a **undocumented** feature in that square brackets are automatically stripped from a material's [`display_name`]() property (if they exist as `name` property delimiters.)
* **Why is this undocumented ?**
* And why does the documentation **erroneously** say "_This should be used when presenting the name in the UI, but the returned name cannot be used as a key in `model.materials`._"
It can be used if they are identical, which is the default when the user creates a material. Ie, the square brackets are meant for internal distributed material names that need translation for the localized builds.

Also, the [`Materials#[]`](https://ruby.sketchup.com/Sketchup/Materials.html#[]-instance_method) method does not automatically delimit key names nor find keys that are bracket delimited without the brackets being explicitly stated in the argument key.

**Comment by Fredosixx** [comment]
I did not know that the `Materials#save_as` would save the material attributes to the skm file.

Then, no need for changing the `display_name`.

So, I can close the issue.

**Comment by DanRathbun** [comment]
> I did not know that the `Materials#save_as` would save the material attributes to the skm file.

**Take note of:**

Ruby API Bug Fixes - SketchUp **2019.2**
> * _Fixed a bug in .skm serialisation where arrays in material attributes were not written out to file_

Ruby API Bug Fixes - SketchUp **2017 MR2**
> * _(Mac) Fixed issue where SKM files lost attribute dictionaries when saved from ‘In Model’ to local library._
